src/createExplorer.ts

```diff
@@ -21,6 +21,15 @@ const MODE_SYNC = 'sync';
 // but it was empty.
 type LoadedFileContent = Config | void;
 
+function nextDirUp(dir: string): string {
+  return path.dirname(dir);
+}
+
+function getExtensionDescription(filepath: string): string {
+  const ext = path.extname(filepath);
+  return ext ? `extension "${ext}"` : 'files without extensions';
+}
+
```

Comment:
I'd prefer not to have an ESLint rule that forces this unnecessary change: I don't think it prevents any errors, and I do think it can sometimes improve readability to take advantage of function hoisting.

Reply:
I think you are referring to not using a function before it is defined? Personally I disagree, but don’t have strong feelings / don’t think it matters for small projects.

This comes from their recommended preset. Should I override?

Reply:
i don't care about stylistic decisions such as this when it's a shared project, but personally i use function hoisting a lot to make my code more readable.  In my opinion it allows for the higher level code to be read first without needing to bother with the implementation of it.

Reply:
I was able to get back at a computer and revert this change. I also set [`'@typescript-eslint/no-use-before-define': 'off'`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-use-before-define.md).